src/test/java/com/exasol/spark/s3/S3CleanupIT.java

```diff
@@ -25,8 +25,16 @@
 import software.amazon.awssdk.services.s3.model.ListObjectsRequest;
 import software.amazon.awssdk.services.s3.model.S3Object;
 
-// For this test suite, we start Spark session with local mode {@code local[*]} with multiple threads for each test unit
-// to force the job end call.
+/**
+ * This suite tests for cleanup processes at the end of Spark jobs.
+ *
+ * For that, we start Spark session with local mode {@code local[*]} with multiple threads for each test unit to force
+ * the job end cleanup calls.
+ *
+ * We use helper {@link TaskFailureStateCounter} class whic is singleton because we will count the number of Spark task
+ * failures concurrently from multiple threads. With this, singleton class variables are shared amond {@code JVM}
```

Comment:
```suggestion
 * failures concurrently from multiple threads. With this, singleton class variables are shared among {@code JVM}
```

Reply:
This doesn't answer, why it needs to be a singleton, if I wouldn't know Spark, I would say, I could use and object.